code/datums/abilities/wraith.dm

```diff
@@ -723,6 +860,1075 @@
+			boutput(W, "something went terribly wrong, call 1-800-CODER")
+			return
+
+		var/obj/spookMarker/marker = new /obj/spookMarker(T)
+		var/list/text_messages = list()
+		text_messages.Add("Would you like to respawn as a plague rat? Your name will be added to the list of eligible candidates and set to DNR if selected.")
+		text_messages.Add("You are eligible to be respawned as a plague rat. You have [src.ghost_confirmation_delay / 10] seconds to respond to the offer.")
+		text_messages.Add("You have been added to the list of eligible candidates. The game will pick a player soon. Good luck!")
+
+		// The proc takes care of all the necessary work (job-banned etc checks, confirmation delay).
+		usr.playsound_local(usr.loc, "sound/voice/wraith/wraithportal.ogg", 50, 0)
+		message_admins("Sending plague rat offer to eligible ghosts. They have [src.ghost_confirmation_delay / 10] seconds to respond.")
+		var/list/datum/mind/candidates = dead_player_list(1, src.ghost_confirmation_delay, text_messages)
+		if (!islist(candidates) || candidates.len <= 0)
+			message_admins("Couldn't set up plague rat ; no ghosts responded. Source: [src.holder]")
+			logTheThing("admin", null, null, "Couldn't set up plague rat ; no ghosts responded. Source: [src.holder]")
+			if (tries >= 1)
+				boutput(W, "No spirits responded. The portal closes.")
+				qdel(marker)
+				return
+			else
+				boutput(W, "Couldn't set up plague rat ; no spirits responded. Trying again in 3 minutes.")
+				qdel(marker)
+				SPAWN(3 MINUTES)
+					make_plague_rat(W, T, tries++)
+			return
+		var/datum/mind/lucky_dude = pick(candidates)
+
+		//add plague rat to master's list is done in /mob/living/critter/wraith/plaguerat/New
+		var/mob/living/critter/wraith/plaguerat/young/P = new /mob/living/critter/wraith/plaguerat/young(T, W)
+		lucky_dude.special_role = ROLE_PLAGUERAT
+		lucky_dude.dnr = 1
+		lucky_dude.transfer_to(P)
+		ticker.mode.Agimmicks |= lucky_dude
+		//Might need to re-add those.
+		//P.ckey = lucky_dude.ckey
+		//P.antagonist_overlay_refresh(1, 0)
+		message_admins("[lucky_dude.key] respawned as a plague rat for [src.holder.owner].")
+		usr.playsound_local(usr.loc, "sound/voice/wraith/ghostrespawn.ogg", 50, 0)
+		logTheThing("admin", lucky_dude.current, null, "respawned as a plague rat for [src.holder.owner].")
+		P.show_antag_popup("plaguerat")
+		boutput(P, "<span class='notice'><b>You have been respawned as a plague rat!</b></span>")
+		boutput(P, "[W] is your master! Eat filth, spread disease and reproduce!")
+		boutput(P, "Obey your master's orders, avoid mouse traps and live the rat life!")
+		qdel(marker)
+
+/datum/targetable/wraithAbility/speak
+	name = "Spirit message"
+	desc = "Telepathically speak to your minions."
+	icon_state = "speak_summons"
+	targeted = 0
+	target_nodamage_check = 1
+	max_range = 1
+	cooldown = 0
+	pointCost = 0
+	proc/ghostify_message(var/message)
+		return message
```

Comment:
?

Reply:
Was a leftover from the previous ghostify message code, removed it